### PR TITLE
Add example usage + API

### PR DESCRIPTION
**Pull request by mixmix**
hey was reviewing https://github.com/ssbc/patchbay/pull/326/files and wanted to understand what you were doing.

This needed more docs, so I wrote some simple docs. If figure if I got it wrong that would be good to know (for both of us!).

Seems like a cool module. Some reflections :
- I presume it behaves in Windows (I've not used `fs.symlink` ... man I wish I knew about that before!)
- the callbacks are a little non-standard - I expected `(err, data)` signature.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,49 @@ The main purpose of this module is to make it easier to integrate
...
 [ssb](https://www.scuttlebutt.nz/).
 
-See also
+## Example usage
+
+```js
+var datSharedFiles = require('dat-shared-files/lib')
+
+// file Object from <input type='file'>
+var datSharedFiles.shareFile(file, (datLink) => {
```
Comment (mixmix) [inline README.md:31]: ooop, I learn already that this should be a `file.path` (from the other PR)

**Comment by mixmix** [comment]
With this sort of PR @arj03 , I expect to just write this but I will likely leave it in your hands to merge, throw out, or pick bits from.

I am interested in replies, I just think it's easier for you to take over making changes to what I wrote yourself

**Comment by mixmix** [comment]
I'm starting a couple of other issues as check items I think could be improved

**Comment by arj03** [comment]
Thanks! I've merged this tidied a bit and released as 1.0, I agree on the signature.

**Pull request merged**